com.unity.shadergraph/Editor/GraphUI/ShaderGraphStencil.cs

```diff
@@ -44,21 +46,44 @@ public override ISearcherFilterProvider GetSearcherFilterProvider()
         }
 
         private Registry.Registry RegistryInstance = null;
+
         public Registry.Registry GetRegistry()
         {
             if (RegistryInstance == null)
             {
-                RegistryInstance = Registry.Default.DefaultRegistry.CreateDefaultRegistry();
+                m_NodeUIHints = new Dictionary<RegistryKey, Dictionary<string, float>>();
+
+                void ReadUIInfo(RegistryKey key, Type type)
+                {
+                    const string uiHintsGetterName = "get_UIHints";
+
+                    var getUiHints = type.GetMethod(uiHintsGetterName);
+                    if (getUiHints != null)
+                    {
+                        m_NodeUIHints[key] = (Dictionary<string, float>)getUiHints.Invoke(null, null);
+                    }
+
+                    // TODO: Get and use UI strings
+                }
+
+                RegistryInstance = Registry.Default.DefaultRegistry.CreateDefaultRegistry(afterNodeRegistered: ReadUIInfo);
             }
+
             return RegistryInstance;
         }
 
+        public bool TryGetUIHints(RegistryKey nodeKey, out IReadOnlyDictionary<string, float> uiHints)
```

Comment:
I think this should just be `GetUIHints` and it should always return a `IReadOnlyDictionary<string, float>` which is empty if there are none. This allows you to not need the bool return and also not require an out var.

Reply:
Makes sense -- the `Try` pattern isn't really adding any meaningful safety, but is making the call site harder to read. Going ahead with `GetUIHints`.